addons/score/spoke/calculate.go

```diff
@@ -0,0 +1,246 @@
+	for _, node := range nodes {
+		if !node.Spec.Unschedulable {
+			for key, value := range node.Status.Allocatable {
+				if allocatable, exist := allocatableList[key]; exist {
+					allocatable.Add(value)
+					allocatableList[key] = allocatable
+				} else {
+					allocatableList[key] = value
+				}
+			}
+		}
+	}
+
+	quantity := allocatableList[resourceName]
+	return quantity.AsApproximateFloat64(), nil
+}
+
+// calculatePodResourceRequest returns the total requested quantity of a certain resource by summing up
+// the request of all containers in the cluster by looping over all pods.
+func (s *ResourceScore) calculatePodResourceRequest(resourceName corev1.ResourceName) (float64, error) {
+	var podRequest float64
+	var podCount int
+
+	list, err := s.podListener.List(labels.Everything())
+	if err != nil {
+		return 0, err
+	}
+
+	for _, pod := range list {
+		for i := range pod.Spec.Containers {
+			container := &pod.Spec.Containers[i]
+			value := s.getRequestForResource(resourceName, &container.Resources.Requests)
+			podRequest += value
+		}
+
+		for i := range pod.Spec.InitContainers {
+			initContainer := &pod.Spec.InitContainers[i]
+			value := s.getRequestForResource(resourceName, &initContainer.Resources.Requests)
+			if podRequest < value {
+				podRequest = value
+			}
+		}
+
+		// If Overhead is being utilized, add to the total requests for the pod
+		if pod.Spec.Overhead != nil && s.enablePodOverhead {
+			if quantity, found := pod.Spec.Overhead[resourceName]; found {
+				podRequest += quantity.AsApproximateFloat64()
+			}
+		}
+		podCount++
+	}
+	return podRequest, nil
+}
+
+// getRequestForResource retrieves the resource request value for the specified resource from the given list of requests.
+// It returns 0 if the requests pointer is nil or if the request for the specified resource is not found.
+func (s *ResourceScore) getRequestForResource(resource corev1.ResourceName, requests *corev1.ResourceList) float64 {
+	if requests == nil {
+		return 0
+	}
+
+	switch resource {
+	case corev1.ResourceCPU:
+		cpuQuantity, found := (*requests)[corev1.ResourceCPU]
+		if !found {
+			return 0
+		}
+		return cpuQuantity.AsApproximateFloat64()
+	case corev1.ResourceMemory:
+		memQuantity, found := (*requests)[corev1.ResourceMemory]
+		if !found {
+			return 0
+		}
+		return memQuantity.AsApproximateFloat64()
+	default:
+		quantity, found := (*requests)[resource]
+		if !found {
+			return 0
+		}
+		return quantity.AsApproximateFloat64()
+	}
+}
+
+// getMaxMinValues retrieves the maximum and minimum values for CPU and memory counts from environment variables.
+// If the environment variables are not set, it uses default values.
+func (s *ResourceScore) getMaxMinValues() ([]float64, error) {
+	envVars := map[string]string{
+		"MAX_CPU_COUNT": defaultMaxCPUCount,
+		"MIN_CPU_COUNT": defaultMinCPUCount,
+		"MAX_MEM_BYTES": defaultMaxMemoryBytes,
+		"MIN_MEM_BYTES": defaultMinMemoryBytes,
+	}
+
+	values := make([]float64, len(envVars))
+
```

Comment:
this function is a bit problematic you need to pass it the given env vars already parsed